typing/typecore.ml

```diff
@@ -3938,8 +3985,9 @@ and type_expect_
       let exp1 = type_statement ~explanation:Sequence_left_hand_side
           env sexp1 in
       let exp2 = type_expect env expected_mode sexp2 ty_expected_explained in
+      let layout = type_layout env exp1.exp_type in
```

Comment:
Yeah, this is gross.  I thought a little about how to improve it, and it's not so obvious to me, so I'm just leaving a CR for now.  I will note that it would be pretty easy to do better if we were happy to restrict exp1 to be representable, but actually I think it makes sense to allow any in this position.